src/formatDate.js

```diff
@@ -50,7 +50,48 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const symbolA = fromFormat[fromFormat.length - 1];
+  const symbolB = toFormat[toFormat.length - 1];
+
+  const newDateArray = date.split(symbolA).join(symbolB).split(symbolB);
+  const newToFormat = toFormat.filter(dateExpression =>
+    dateExpression !== dateExpression[dateExpression.length - 1]);
+  const newFromFormat = fromFormat.filter(dateExpression =>
+    dateExpression !== dateExpression[dateExpression.length - 1]);
+
+  const obj = {};
+  const objFormatted = {};
+
+  for (let i = 0; i < newFromFormat.length; i++) {
+    obj[newFromFormat[i]] = newDateArray[i];
+  }
+
+  for (let i = 0; i < newToFormat.length; i++) {
```

Comment:
Refactor the nested loops into a single loop by directly mapping the 'filteredToFormat' array to the corresponding values in the 'dateObj'